fast 3D tracking using galvo scan/descan

dpshepherd opened the pull request:
Start of branch for fast galvo imaging. A lot of quality of life changes to do here, but the initial commit takes data and reconstructs it into a BDV H5 file.

dpshepherd committed:
initial upload of merfish probe design (in progress)

dpshepherd committed:
Update README.md

dpshepherd committed:
Update README.md

dpshepherd committed:
Update README.md

dpshepherd committed:
Update readout + encoding probe generation and metadata

dpshepherd committed:
Minor updates

dpshepherd committed:
Update README.md

dpshepherd committed:
Update README.md

dpshepherd committed:
Primer design and general workflow updates

dpshepherd committed:
Merge branch 'merfish' of https://github.com/QI2lab/OPM into merfish

dpshepherd committed:
Finish primer design; Initial commit for pycro-manager

dpshepherd committed:
Testing ready probe dseign code

Rename script, finished primer design, added sequential design, created file saving, fixed reverse complement issue with readout sequences, added T7 and final probe sequence assembly for ordering. This version if ready for in silcio testing against human knowledge and prior probe sets.

dpshepherd committed:
Update README.md

dpshepherd committed:
Update README.md

dpshepherd committed:
Update README.md

dpshepherd committed:
Initiial work on pycromanager scripts

dpshepherd committed:
Refactor to include fluidics control with pycromanager control

dpshepherd committed:
Update run_opm.py

dpshepherd committed:
Updated pycromanager stage scan control and reconstruction

dpshepherd committed:
Changes to support writing translations into bigstitcher file during creation

dpshepherd committed:
Update to acquisition and reconstruction code to include flat-fielding and iterative rounds

dpshepherd committed:
More automated stitching updates

dpshepherd committed:
Further updates for full automation

dpshepherd committed:
Push current stage and galvo control/recon code

dpshepherd committed:
Clean up

dpshepherd committed:
clean up more

ptbrown1729 committed:
added localization script

dpshepherd committed:
Create view_galvo_scan.py

ptbrown1729 committed:
first draft of localization

ptbrown1729 committed:
Merge branch 'galvo' of https://github.com/QI2lab/OPM into galvo

dpshepherd commented on localization/localization_data.py:
```diff
@@ -0,0 +1,243 @@
+# ###############################
+# interpolate images so are on grids in coverslip coordinate system and plot all results
+# ###############################
+xi, yi, zi, imgs_unskew = localize.interp_opm_data(imgs_filtered, dc, dstage, theta, mode="ortho-interp")
```
@ptbrown1729 Out of curiosity, is there a difference between the orthogonal interpolation that you wrote and that I wrote?

Does it make sense to refactor and get OPM specific functions into their own library?

ptbrown1729 replied:
@dpshepherd mine was just a way to build my understanding. I am using it in some of the localization tests, but ultimately it should be removed.

I agree it might make sense to put any OPM specific stuff in a separate place than localization.py

Right now the organization is not great. I pushed the change as I started trying to localize the full image and wanted to work on a different computer

dpshepherd replied:
Sounds good. Let's brainstorm how to better organize, I agree what I have is all over the place. Things on my end have become really messy recently with all of the changes.

ptbrown1729 replied:
I just meant the stuff I pushed was disorganized

ptbrown1729 committed:
Corrected interpolation for case where the stage step is not exactly a
multiple of the pixel size * cos(theta). First working version of
diffusive bead tracking localization and tracking, using home built
localization and pytrack for tracking. Changed pycromanager data loading
scheme to work with this large dataset.
minor changes

dpshepherd committed:
updates

dpshepherd committed:
added script to run galvo scan using HCImage

ptbrown1729 committed:
updates to localization scripts

ptbrown1729 committed:
Merge branch 'galvo' of https://github.com/QI2lab/OPM into galvo

ptbrown1729 committed:
updated localization scripts

dpshepherd committed:
Merge branch 'galvo' of https://github.com/QI2lab/OPM into galvo

dpshepherd committed:
Update run_opm_galvoscan_hcimage.py

ptbrown1729 committed:
updated tracking/localization

ptbrown1729 committed:
Merge branch 'galvo' of https://github.com/QI2lab/OPM into galvo

dpshepherd committed:
Updated scan control

ptbrown1729 committed:
(1) updated script to work with multiple color laser sequences (2) added command to ensure the galvo starts from the correct position, otherwise can run into problems where the galvo has to make a large step at the first frame, which it cannot do fast enough

dpshepherd committed:
Breaking changes to galvo scan for metadata

ptbrown1729 committed:
various updates to localization including (1) switching to max filter thresholding, which speeds things up (2) corrected difference of gaussian filtering (3) GPU filtering using cupy, not had to download and build 9.0.0rc1 from their github site to get the new function cupyx.scipy.signal.fftconvolve (4) update localization explorer tool in matplotlib. Various other updates removing unused functions or functionality

ptbrown1729 committed:
Merge branch 'galvo' of https://github.com/QI2lab/OPM into galvo

dpshepherd committed:
Push most recent galvo reconstruction with changes made for HCImage or Pycromanager

ptbrown1729 committed:
minor updates

ptbrown1729 committed:
Merge branch 'galvo' of https://github.com/QI2lab/OPM into galvo

ptbrown1729 committed:
(1) updated functions to read hcimage or pycromanager style datasets (2) improved localization display (3) renamed some files

ptbrown1729 committed:
fixed load_dataset.py for linux

dpshepherd committed:
Further updates to control and recon code

dpshepherd committed:
Merge branch 'galvo' of https://github.com/QI2lab/OPM into galvo

dpshepherd commented:
I'm closing this, since we have merged the hardware and reconstruction changes. New work is taking place in the "localization" branch on 3D localization and tracking.
